cmd/agent/subcommands/run/command.go

```diff
@@ -342,6 +342,10 @@ func getSharedFxOption() fx.Option {
 		remoteconfig.Bundle(),
 		fx.Provide(tagger.NewTaggerParamsForCoreAgent),
 		tagger.Module(),
+		autodiscoveryimpl.Module(),
+		fx.Provide(func(ac autodiscovery.Component) optional.Option[autodiscovery.Component] {
+			return optional.NewOption[autodiscovery.Component](ac)
+		}),
```

Comment:
Can you move this function to autodiscovery module like [here](https://github.com/DataDog/datadog-agent/pull/23388/files#diff-ce621907cd6fb881813ce35acaca2f0a835a8dd9fe8ccfb58e3a1b51b168626aR132-R136).